apps/server/src/modules/account/uc/account.uc.spec.ts

```
@@ -102,1130 +57,3073 @@ describe('AccountUc', () => {
+					roles: [new Role({ name: RoleName.STUDENT, permissions: [] })],
+				});
+
+				userRepo.findById.mockResolvedValueOnce(mockTeacherUser).mockResolvedValueOnce(mockAdminUser);
+				userRepo.findById.mockResolvedValueOnce(mockStudentUser).mockResolvedValueOnce(mockOtherStudentUser);
+				userRepo.findById.mockResolvedValueOnce(mockStudentUser).mockResolvedValueOnce(mockTeacherUser);
+
+				return { mockTeacherUser, mockAdminUser, mockStudentUser, mockOtherStudentUser };
+			};
+			it('should throw ForbiddenOperationError', async () => {
+				const { mockTeacherUser, mockAdminUser, mockStudentUser, mockOtherStudentUser } = setup();
+				await expect(
+					accountUc.searchAccounts(
+						{ userId: mockTeacherUser.id } as ICurrentUser,
+						{ type: AccountSearchType.USER_ID, value: mockAdminUser.id } as AccountSearchQueryParams
+					)
+				).rejects.toThrow(ForbiddenOperationError);
+
+				await expect(
+					accountUc.searchAccounts(
+						{ userId: mockStudentUser.id } as ICurrentUser,
+						{ type: AccountSearchType.USER_ID, value: mockOtherStudentUser.id } as AccountSearchQueryParams
+					)
+				).rejects.toThrow(ForbiddenOperationError);
+
+				await expect(
+					accountUc.searchAccounts(
+						{ userId: mockStudentUser.id } as ICurrentUser,
+						{ type: AccountSearchType.USER_ID, value: mockTeacherUser.id } as AccountSearchQueryParams
+					)
+				).rejects.toThrow(ForbiddenOperationError);
+			});
+		});
+
+		describe('When search type is unknown', () => {
+			const setup = () => {
+				const mockSchool = schoolFactory.buildWithId();
+
+				const mockSuperheroUser = userFactory.buildWithId({
+					school: mockSchool,
+					roles: [
+						new Role({
+							name: RoleName.SUPERHERO,
+							permissions: [Permission.TEACHER_EDIT, Permission.STUDENT_EDIT],
+						}),
+					],
+				});
+
+				userRepo.findById.mockResolvedValueOnce(mockSuperheroUser);
+
+				return { mockSuperheroUser };
+			};
+			it('should throw', async () => {
```

should throw invalid search error?